views/templates/front/payment.tpl

```diff
@@ -23,43 +23,80 @@
  * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
  *}
 
-<iframe id="comfino-paywall-container" src="{$paywall_api_url}" referrerpolicy="strict-origin" loading="lazy" class="comfino-paywall" scrolling="no" onload="ComfinoPaywallFrontend.onload(this, '{$paywall_options.platformName|escape:"htmlall":"UTF-8"}', '{$paywall_options.platformVersion|escape:"htmlall":"UTF-8"}')"></iframe>
+<div id="comfino-iframe-container"></div>
 <script>
-    if (ComfinoPaywallFrontend.isInitialized()) {
-        Comfino.init();
-    } else {
-        window.Comfino = {
-            paywallOptions: {$paywall_options|@json_encode nofilter},
-            init: () => {
-                ComfinoPaywallFrontend.init(
-                    document.querySelector('input[data-module-name="comfino"]'),
-                    document.getElementById('comfino-paywall-container'),
-                    Comfino.paywallOptions
+    window.Comfino = {
+        paywallOptions: {$paywall_options|@json_encode nofilter},
+        init: () => {
+            var iframe = document.createElement('iframe');
+            iframe.id = 'comfino-paywall-container';
+            iframe.className = 'comfino-paywall';
+            iframe.onload = function () {
+                ComfinoPaywallFrontend.onload(
+                    this,
+                    '{$paywall_options.platformName|escape:"htmlall":"UTF-8"}',
+                    '{$paywall_options.platformVersion|escape:"htmlall":"UTF-8"}'
                 );
-            }
-        }
+            };
+            iframe.src = '{$paywall_api_url}';
+            iframe.referrerPolicy = 'strict-origin';
+            iframe.loading = 'lazy';
+            iframe.scrolling = 'no';
 
-        Comfino.paywallOptions.onUpdateOrderPaymentState = (loanParams) => {
-            ComfinoPaywallFrontend.logEvent('updateOrderPaymentState PrestaShop', 'debug', loanParams);
+            let frontendInitElement = document.querySelector('input[data-module-name="comfino"]');
 
-            let offersUrl = '{$offers_url}'.replace(/&amp;/g, '&');
-            let urlParams = new URLSearchParams({ loan_type: loanParams.loanType, loan_term: loanParams.loanTerm });
+            if ('priceModifier' in frontendInitElement.dataset) {
+                let priceModifier = parseFloat(frontendInitElement.dataset.priceModifier);
 
-            offersUrl += (offersUrl.indexOf('?') > 0 ? '&' : '?') + urlParams.toString();
+                if (!Number.isNaN(priceModifier)) {
+                    iframe.src += ('&priceModifier=' + priceModifier);
+                }
+            }
```

Comment:
Co robi `priceModifer`?? Chodzi o `,` lub `.`? Nie dało rady tego obsłużyć po stronie iframe?

Reply:
Chodzi o dynamiczne doliczanie prowizji do płatności Comfino na poziomie JavaScript (kolejna zajefajna funkcja któregoś tam OPC: https://test.boboexpert.pl). To wszystko dzieje się w locie na froncie, stąd takie akrobacje. Średnio mi się to podoba, ale cóż...

"Opis problemu: Mam moduł do PrestaShop 1.7 który w one page checkout 
dodaje produkt typu "prowizja" do koszyka przy pomocy JavaScript. Mam 
też moduł comfino. W module comfino wartość zakupów, czyli kwota kredytu 
nie aktualizuje się gdy dodaje produkt "prowizja" przy pomocy JavaScript 
(w one page checkout). Proszę o pomoc. Chodzi o widok checkout. Mamy 
zainstalowany moduł, który powoduje, że po dodaniu do koszyka produktu 
przecenionego i wyborze płatności "Comfino - Raty 0%" dodaje się inny 
produkt typu "prowizja". Operacja jest wykonywana przy pomocy JavaScript.

Jak widzimy kwota zamówienia równa się 1 700,34 zł

Niestety wartość zakupów w metodzie płatności Comfino równa się kwocie 
zamówienia bez prowizji. Nie dodaje się wartość prowizji do kwoty 
zamówienia w Comfino."